File: golang/internal/models/exchange.go
package models

import (
	"github.com/ethereum/go-ethereum/common"
	ethTypes "github.com/ethereum/go-ethereum/core/types"
)

// ExchangeReq exchange response structure
type ExchangeReq struct {
	UUID    string           `json:"uuid"`
	TxHash  common.Hash      `json:"tx_hash"`
	Receipt ethTypes.Receipt `json:"receipt"`
	Intent  ExchangeIntent   `json:"intent"`
}

// PairAsset asset pair structure
type PairAsset struct {
	Type    string `json:"type"`
	AssetID string `json:"id"`
	Amount  uint   `json:"amount"`
}

// ExchangeIntent exchange intent structure
type ExchangeIntent struct {
	Type   string      `json:"type"`
	Method string      `json:"method"`
	From   []PairAsset `json:"from"`
	To     []PairAsset `json:"to"`
}

File: csharp/SampleGameBackend/Models/Exchange.cs
using System.Text.Json.Serialization;

namespace SampleGameBackend.Models
{
    public class ExchangeResultRequest
    {
        [JsonPropertyName("uuid")]
        public string Uuid { get; set; } = string.Empty;

        [JsonPropertyName("tx_hash")]
        public string TxHash { get; set; } = string.Empty;

        [JsonPropertyName("receipt")]
        public object Receipt { get; set; } = new();

        [JsonPropertyName("intent")]
        public ExchangeIntent Intent { get; set; } = new();
    }

    public class PairToken
    {
        [JsonPropertyName("id")]
        public string TokenId { get; set; } = string.Empty;

        [JsonPropertyName("amount")]
        public uint Amount { get; set; }
    }

    public class PairAsset
    {
        [JsonPropertyName("id")]
        public string AssetId { get; set; } = string.Empty;

        [JsonPropertyName("amount")]
        public uint Amount { get; set; }
    }

    public class ExchangeIntent
    {
        [JsonPropertyName("type")]
        public string Type { get; set; } = string.Empty;

        [JsonPropertyName("method")]
        public string Method { get; set; } = string.Empty;

        [JsonPropertyName("from")]
        public List<PairAsset> From { get; set; } = new();

        [JsonPropertyName("to")]
        public List<PairAsset> To { get; set; } = new();
    }
} 